File: TodoAssessment/ClientApp/src/app/interfaces/TodoEntry.ts
type GUID = string & { isGuid: true };
function guid(guid: string): GUID {
	return guid as GUID; // maybe add validation that the parameter is an actual guid ?
}

export interface TodoEntry {
	id: GUID;
	title: string;
	createDate: Date;
	dueDate: Date;
	status: number;
}

File: Shared.Models/TodoEntry.cs
using System;
namespace Shared.Models
{
    public class TodoEntry
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime DueDate { get; set; }
        public TaskStatus Status { get; set; }
    }
}

